Row.c

```diff
@@ -190,94 +190,112 @@ RowField RowField_keyAt(const Settings* settings, int at) {
    return COMM;
 }
 
-void Row_printBytes(RichString* str, unsigned long long number, bool coloring) {
+void Row_printKBytes(RichString* str, unsigned long long number, bool coloring) {
    char buffer[16];
    int len;
 
-   int largeNumberColor = coloring ? CRT_colors[LARGE_NUMBER] : CRT_colors[PROCESS];
-   int megabytesColor = coloring ? CRT_colors[PROCESS_MEGABYTES] : CRT_colors[PROCESS];
-   int gigabytesColor = coloring ? CRT_colors[PROCESS_GIGABYTES] : CRT_colors[PROCESS];
-   int shadowColor = coloring ? CRT_colors[PROCESS_SHADOW] : CRT_colors[PROCESS];
-   int baseColor = CRT_colors[PROCESS];
+   int color = CRT_colors[PROCESS];
+   int nextUnitColor = CRT_colors[PROCESS];
 
-   if (number == ULLONG_MAX) {
-      //Invalid number
-      RichString_appendAscii(str, shadowColor, "  N/A ");
+   int colors[4] = {
+      [0] = CRT_colors[PROCESS],
+      [1] = CRT_colors[PROCESS_MEGABYTES],
+      [2] = CRT_colors[PROCESS_GIGABYTES],
+      [3] = CRT_colors[LARGE_NUMBER]
+   };
+
+   if (number == ULLONG_MAX)
+      goto invalidNumber;
+
+   if (coloring) {
+      color = colors[0];
+      nextUnitColor = colors[1];
+   }
+
+   if (number < 1000) {
+      // Plain number, no markings
+      len = xSnprintf(buffer, sizeof(buffer), "%5u ", (unsigned int)number);
+      RichString_appendnAscii(str, color, buffer, len);
+      return;
+   }
+   if (number < 100000) {
+      // 2 digits for M, 3 digits for K
+      len = xSnprintf(buffer, sizeof(buffer), "%2u", (unsigned int)(number / 1000));
+      RichString_appendnAscii(str, nextUnitColor, buffer, len);
+      len = xSnprintf(buffer, sizeof(buffer), "%03u ", (unsigned int)(number % 1000));
+      RichString_appendnAscii(str, color, buffer, len);
       return;
    }
 
-   number /= ONE_K;
+   // 100000 KiB (97.6 MiB) or greater. A unit prefix would be added.
+   const size_t maxUnitIndex = (sizeof(number) * CHAR_BIT - 1) / 10 + 1;
+   const bool canOverflow = maxUnitIndex >= ARRAYSIZE(unitPrefixes);
+
+   size_t i = 1;
+   int prevUnitColor;
+   // Convert KiB to (1/100) of MiB
+   unsigned long long hundredths = (number / 256) * 25 + (number % 256) * 25 / 256;
```

Comment:
I'd prefer `uint64_t` over those clunky `unsigned long long int`s (type-ref-wise) … ;-)

Reply:
That would change the API/prototype of `Row_printKBytes`.
(I know `uint64_t` is a synonym of `unsigned long long` in most 64-bit OSes, but in the standard's sense, `unsigned long long` is "64 bits or more" and a cast to `uint64_t` may serve as a type demotion.)

Reply:
As `uint64_t` is an alias for `unsigned long long int`, that's mostly cosmetic. I can live with that …